atom/src/catom.cpp

```diff
@@ -445,29 +445,36 @@ CAtom_getstate( CAtom* self )
 }
 
 PyObject*
-CAtom_setstate( CAtom* self, PyObject* args )
+CAtom_setstate( CAtom* self, PyObject* state )
 {
-    if( PyTuple_GET_SIZE( args ) != 1 )
-        return cppy::type_error( "__setstate__() takes exactly one argument" );
-    PyObject* state = PyTuple_GET_ITEM( args, 0 );
-    cppy::ptr itemsptr = PyMapping_Items(state);
-    if ( !itemsptr )
-        return 0;
-    cppy::ptr selfptr(pyobject_cast(self), true);
+    if ( !PyMapping_Check(state) )
+        return cppy::type_error("__setstate__ requires a mapping");
 
     // If the -f key is present freeze the object
-    bool frozen = PyMapping_HasKey(state, atom_flags);
+#if PY_VERSION_HEX >= 0x030D0000
```

Comment:
I checked the documentation of this function but I do not understand why we need to use it here. Since we do not have the CI failure history, can you explain @frmdstryr ?

Reply:
I think it's just a warning but pytest still makes it fail: See the log here https://github.com/nucleic/atom/actions/runs/12874904293/job/35895250664 . 

The HasKeyWithError version makes more sense anyways.

Reply:
Thanks. It really shows what a bad job PyMapping_Check does. I will merge.